src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -709,14 +709,9 @@ protected override bool HasImplicitTupleConversion(BoundExpression source, TypeS
             var tupleExpression = (BoundTupleLiteral)source;
             var arguments = tupleExpression.Arguments;
 
-            // unwrap tuple to its underlying
-            if (destination.IsTupleType)
-            {
-                destination = ((TupleTypeSymbol)destination).UnderlyingTupleType;
-            }
-
-            // check if underlying type is actually a possible underlying type for a tuple of given arity
-            if(!Compilation.IsWellKnownTupleType(destination, arguments.Length))
+            // check if the type is actually compatible type for a tuple of given cardinality
+            int cardinality;
+            if (!destination.IsTupleCompatible(out cardinality) || cardinality != arguments.Length)
```

Comment:
I'll consider this for the next change.

---
In reply to: [61792125](https://github.com/dotnet/roslyn/pull/11006#discussion_r61792125) [](ancestors = 61792125)